### PR TITLE
4 store the maximum precision with the shared table configuration

### PR DESCRIPTION
**Pull request by alexeiakimov**

### CODE REVIEW

File: server/src/universal/conf/delta-sharing-server.yaml.template [hunk 1]
```diff
@@ -8,8 +8,10 @@ shares:
     tables:
     - name: "table1"
       location: "s3a://<bucket-name>/<the-table-path>"
+      precision: 0.1
```
Comment (alexeiakimov) [inline server/src/universal/conf/delta-sharing-server.yaml.template:11]: This is how the precision is added to the configuration. @cugni is it what we need?

File: server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala [hunk 2]
```diff
@@ -159,12 +159,15 @@ class DeltaSharedTable(
           } else {
             selectedFiles
           }
-        val sampleFiles =
-          if (sampleHint.nonEmpty) {
-            Sampler.sample(filteredFiles, sampleHint.get)
-          } else {
-            filteredFiles
-          }
+        val (precisionFrom, precisionTo) = sampleHint match {
```
Comment (alexeiakimov) [inline server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:162]: Here the precision defined in the configuration is combined with the precision range specified in the request. If there is no sample hint in the request, then only predefined precision is used.

Comment (cugni) [reply server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:162]: Perfect, and it takes the min of them. Cool, so this should work also when we build a "Qbeast aware" client.

**Review by cugni — APPROVED** [review]
It seems all good! great job!

**Pull request merged**